service/controller/app/v1/resource/chart/current.go

```diff
@@ -29,10 +29,11 @@ func (r *Resource) GetCurrentState(ctx context.Context, obj interface{}) (interf
 
 	chart, err := cc.G8sClient.ApplicationV1alpha1().Charts(r.chartNamespace).Get(name, metav1.GetOptions{})
 	if errors.IsNotFound(err) {
-		r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("did not find chart %#q", name))
+		r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("did not find chart %#q in namespace %#q", name, r.chartNamespace))
 		return nil, nil
 	} else if err != nil {
-		return nil, microerror.Mask(err)
+		r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("could not find chart %#q in namespace %#q due to an error: %#q", name, r.chartNamespace, err))
+		return nil, nil
```

Comment:
I don't think this is right. When there are unexpected errors we should return them to the caller. We only seem to have problems with errors that we kind of expect, so we should match them and handle them specifically. The "not found error" is such a special case already. We only need to add another special case.

Reply:
In the original issue, the operator got an error as below. ```x509: certificate is valid for ingress.local, not api.hd39g.k8s.ginger.eu-central-1.aws.gigantic.io``` I'm not sure which cases we should not report as an error in order to not prevent namespace deletion.

Reply:
Well, this case at least I think. We can use something like this here: https://godoc.org/github.com/giantswarm/errors/tenant.

Reply:
Wow thanks! Great hint!